### PR TITLE
Starter code for five modules

### PR DESCRIPTION
**Pull request by wheelmac92**
Nothing too advanced, mostly just implemented proper error codes.
SceError properly prints error messages, without opening a window.
This can be updated later to use IMGUI, or whichever window handler we want to use

### CODE REVIEW

**Comment by Inori** [comment]
I'm sorry but this pull request is hard to merge, its' style is far from our code style standard.
If you're not sure about the style, just look how others write code.

And apart from that, for a module which is complex enough, we'll use external classes to handle it, just like [we did in the ScePad module](https://github.com/Inori/GPCS4/blob/gamepad/GPCS4/SceModules/ScePad/sce_pad.cpp)

So I'm going to discard this pr, sorry again.

**Comment by Inori** [comment]
Another thing is that, implement error codes seems not that meaningful, because we need these stub functions to return `SCE_OK` in most cases, to let the program go on, even if it does nothing.

Maybe you can try other things listed on the TODO list :)

**Comment by mailwl** [comment]
return 0 not always good idea, some games just exit without visible reasons

File: GPCS4/SceModules/SceAudioOut/sce_audioout.cpp [hunk 1]
```diff
@@ -11,56 +11,200 @@
 // library: libSceAudioOut
 //////////////////////////////////////////////////////////////////////////
 
+static bool init = false;
```
[the remark this answers is not in the document]
Comment (wheelmac92) [reply GPCS4/SceModules/SceAudioOut/sce_audioout.cpp]: Done

File: GPCS4/SceModules/SceAudioOut/sce_audioout.cpp [hunk 2]
```diff
@@ -11,56 +11,200 @@
...
+	}
+
+	else if (len < 0)
+	{
+		returnVal = SCE_AUDIO_OUT_ERROR_INVALID_SIZE;
+	}
+
+	else
+	{
+		LOG_SCE_DUMMY_IMPL();
```
[the remark this answers is not in the document]
Comment (wheelmac92) [reply GPCS4/SceModules/SceAudioOut/sce_audioout.cpp]: fixed

**Comment by wheelmac92** [comment]
I spent the last hour improving the code. Is this PR able to be merged?

**Comment by VelocityRa** [comment]
get rid of the project file changes (looks like line endings were changed and you pushed it)

**Comment by VelocityRa** [comment]
and learn to force push... there are 32 commits here for no reason
squash everything to one

**Comment by wheelmac92** [comment]
I deleted the fork, I'm making a new fork, I messed something up badly

**Comment by mailwl** [comment]
how do you think, Sony not crash your PR?

вт, 3 дек. 2019 г. в 20:09, wheelmac92 <notifications@github.com>:

**Comment by wheelmac92** [comment]
@mailwl I messed up the PR myself, it was my mistake. Here is the new PR
https://github.com/Inori/GPCS4/pull/26

**Pull request closed without merging**